install/install.queries.php

```diff
@@ -149,98 +142,143 @@ function encryptFollowingDefuse($message, $ascii_key)
 if (empty($session_db_encoding) === true) {
     $session_db_encoding = 'utf8';
 }
-
 $superGlobal->put('CPM', 1, 'SESSION');
 
-if (null !== $post_type) {
-    switch ($post_type) {
+// Prepare POST variables
+/*
+$inputData['type'] = filter_input(INPUT_POST, 'type', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+$post_data = filter_input(INPUT_POST, 'data', FILTER_SANITIZE_FULL_SPECIAL_CHARS, FILTER_FLAG_NO_ENCODE_QUOTES);
+$post_activity = filter_input(INPUT_POST, 'activity', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+$post_task = filter_input(INPUT_POST, 'task', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+$inputData['index'] = filter_input(INPUT_POST, 'index', FILTER_SANITIZE_NUMBER_INT);
+$inputData['multiple'] = filter_input(INPUT_POST, 'multiple', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+$post_db = filter_input(INPUT_POST, 'db', FILTER_SANITIZE_FULL_SPECIAL_CHARS);
+*/
+$postValues = [
+    'type' => filter_input(INPUT_POST, 'type', FILTER_SANITIZE_FULL_SPECIAL_CHARS),
+    'data' => filter_input(INPUT_POST, 'data', FILTER_SANITIZE_FULL_SPECIAL_CHARS, FILTER_FLAG_NO_ENCODE_QUOTES),
+    'activity' => filter_input(INPUT_POST, 'activity', FILTER_SANITIZE_FULL_SPECIAL_CHARS),
+    'task' => filter_input(INPUT_POST, 'task', FILTER_SANITIZE_FULL_SPECIAL_CHARS),
+    'index' => filter_input(INPUT_POST, 'index', FILTER_SANITIZE_NUMBER_INT),
+    'multiple' => filter_input(INPUT_POST, 'multiple', FILTER_SANITIZE_FULL_SPECIAL_CHARS),
+    'db' => filter_input(INPUT_POST, 'db', FILTER_SANITIZE_FULL_SPECIAL_CHARS),
```

Comment:
`FILTER_SANITIZE_FULL_SPECIAL_CHARS` is not intended to protect against `SQL injection` but rather against `XSS`.
It would seem more secure to use [mysqli_real_escape_string](https://www.php.net/manual/fr/mysqli.real-escape-string.php).

```php
$postValues['db'] = array_map(function($value) use ($bddConnection) {
    return mysqli_real_escape_string($bddConnection, $value);
}, $_POST['db']);
```

Reply:
Yes but here $_POST['db'] contains the requested info to open the mysqli connection.
So before they are needed to initiate $bddConnection.

mysqli_real_escape_string  could indeed being used to secure $_POST['data'] and others